pkg/v3/hooks/build/coordinate_block.go

```diff
@@ -0,0 +1,31 @@
+package build
+
+import (
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg/v3"
+	"github.com/smartcontractkit/ocr2keepers/pkg/v3/flows"
+	"github.com/smartcontractkit/ocr2keepers/pkg/v3/instruction"
+)
+
+const (
+	ShouldCoordinateBlock instruction.Instruction = "should coordinate block"
+	DoCoordinateBlock     instruction.Instruction = "do coordinate block"
+)
+
+type BuildHook func(*ocr2keepers.AutomationObservation, instruction.InstructionStore, ocr2keepers.MetadataStore, flows.ResultStore) error
+
+type coordinateBlockHook struct{}
+
+func NewCoordinateBlockHook() *coordinateBlockHook {
+	return &coordinateBlockHook{}
+}
+
+func (h *coordinateBlockHook) RunHook(obs *ocr2keepers.AutomationObservation, instructionStore instruction.InstructionStore, metadataStore ocr2keepers.MetadataStore, resultStore flows.ResultStore) error {
+	if instructionStore.Has(ShouldCoordinateBlock) {
+		obs.Instructions = append(obs.Instructions, ShouldCoordinateBlock)
+	} else if instructionStore.Has(DoCoordinateBlock) {
+		obs.Metadata["blockHistory"] = metadataStore.GetBlockHistory()
+		instructionStore.Delete(ShouldCoordinateBlock)
```

Comment:
I don't think we will get to this point if there is a should in the instruction store. The first if condition will trigger first. I think the pre-build hook should be responsible for removing the should and adding the do based on what is in the outcome. We probably shouldn't do writes to the stores within the build hooks to maintain a one-way data flow. This will also maintain state consistency across multiple build hooks.